libcaja-private/caja-open-with-dialog.c

```diff
@@ -249,12 +249,18 @@ add_or_find_application (CajaOpenWithDialog *dialog)
 
     if (app == NULL)
     {
-        message = g_strdup_printf (_("Could not add application to the application database: %s"), error->message);
+        if (error)
+        {
+            message = g_strdup_printf (_("Could not add application to the application database: %s"), error->message);
+            g_error_free (error);
+        }
+        else
+            message = _("unknown error");
```

Comment:
This might not be clear enough as an error message.  I'd rather have the same one as in the normal case, just with the specific detail using this "unknown error".
Something like this:
```diff
-        message = g_strdup_printf (_("Could not add application to the application database: %s"), error->message);
+        message = g_strdup_printf (_("Could not add application to the application database: %s"), error ? error->message : _("unknown error"));
```
If you don't like ternary, you could easily use an extra variable:
```diff
-        message = g_strdup_printf (_("Could not add application to the application database: %s"), error->message);
+        const gchar *detail;
+
+        if (error)
+            detail = error->message;
+        else
+            detail = _("unknown error");
+        message = g_strdup_printf (_("Could not add application to the application database: %s"), detail);
```

Reply:
done